### PR TITLE
perf(material-experimental/mdc-progress-bar): Use class for MDC adapter

### PR DESCRIPTION
**Pull request by ngwattcos**
Updates mdc-progress-bar to use a class-based adapter, which improves performance, just like in:
#19980

Hopefully, test failures will be resolved after this PR is merged:
https://github.com/material-components/material-components-web/pull/6256

### CODE REVIEW

**Comment by mmalerba** [comment]
Are the checkbox changes here intentionally or was that some kind of bad rebase?

File: src/material-experimental/mdc-progress-bar/progress-bar.ts [hunk 1]
```diff
@@ -28,6 +28,44 @@ import {Subscription, fromEvent, Observable} from 'rxjs';
...
+  }
+
+  setPrimaryBarStyle(styleProperty: string, value: string) {
+    (this._delegate._primaryBar.style as any)[styleProperty] = value;
+  }
+
+  setBufferBarStyle(styleProperty: string, value: string) {
+    (this._delegate._bufferBar.style as any)[styleProperty] = value;
+  }
+
```
Comment (mmalerba) [inline src/material-experimental/mdc-progress-bar/progress-bar.ts:66]: nit: remove extra blank line

**Comment by ngwattcos** [comment]
@mmalerba I accidentally left changes in checkbox and radio on my branch - should be gone now.

**Comment by mmalerba** [comment]
still seeing checkbox and radio changes

**Comment by ngwattcos** [comment]
@mmalerba I accidentally committed instead of amended the commit. Should be fixed now!

File: src/material-experimental/mdc-progress-bar/progress-bar.ts [hunk 2]
```diff
@@ -28,6 +28,43 @@ import {Subscription, fromEvent, Observable} from 'rxjs';
 import {filter} from 'rxjs/operators';
 import {Directionality} from '@angular/cdk/bidi';
 
+/** @docs-private */
```
Comment (mmalerba) [inline src/material-experimental/mdc-progress-bar/progress-bar.ts:31]: not needed since this class isn't exported

**Comment by mmalerba** [comment]
It looks like there are some failing tests

**Comment by andrewseguin** [comment]
Closing due to inactivity

**Pull request closed without merging**